packages/fast-components-styles-msft/src/utilities/density.ts

```diff
@@ -1,6 +1,146 @@
+import { applyTypeRampConfig, TypeRamp, typeRamp } from "../utilities/typography";
 import { DesignSystem, withDesignSystemDefaults } from "../design-system";
+import { CSSRules } from "@microsoft/fast-jss-manager";
 import { toPx } from "@microsoft/fast-jss-utilities";
 
+/**
+ * Returns the type ramp adjustment from the default setting. Collapses
+ * the fact that a single type size is used for multiple densities.
+ *
+ * @param designSystem The design system config
+ */
+export function densityToTypeOffset(designSystem: DesignSystem): number {
+    const offset: number =
+        designSystem.density >= 2 ? 1 : designSystem.density <= -2 ? -1 : 0;
+    return offset;
+}
+
+/**
+ * Returns the component height formatted in the provided unit or px by default.
+ *
+ * @param lines The logical number of lines the component takes, typically 1.
+ * @param unit The unit of measurement; px by default.
+ */
+export function height(
+    lines: number = 1,
+    minusBorder: number = 0,
+    unit?: string
+): (config: DesignSystem) => string {
+    return (config: DesignSystem): string => {
+        const designSystem: DesignSystem = withDesignSystemDefaults(config);
+        const value: number = heightNumber(lines, minusBorder)(designSystem);
+        return typeof unit === "string" ? `${value}${unit}` : toPx(value);
+    };
+}
+
+/**
+ * Returns the component height as a number.
+ *
+ * @param lines The logical number of lines the component takes, typically 1.
+ * @param minusBorder Any border that should be removed from the overall height.
+ */
+export function heightNumber(
+    lines: number,
+    minusBorder: number = 0
+): (config: DesignSystem) => number {
+    return (config: DesignSystem): number => {
+        const designSystem: DesignSystem = withDesignSystemDefaults(config);
+        const value: number =
+            (designSystem.defaultHeightMultiplier + designSystem.density) *
+                designSystem.designUnit *
+                lines -
+            minusBorder;
+        return value;
+    };
+}
+
+export function minHeight(
```

Comment:
These two seem like pretty generic names - can you elboarate on how these should be used?

Reply:
Admittedly they are carryovers from the previous height utility, but turned into a formula based on the design system instead of fixed values. Currently the only place they are used are in actual min/max height css attributes. There's not much point to that though, as those places are not variable-sized components. They are always used with an explicit height. It's possible we don't need this notion and should just remove any setting of min/max based on density.

Reply:
I removed these.